crates/bevy_ecs/src/bundle.rs

```diff
@@ -393,7 +435,10 @@ impl BundleInfo {
             let component_id = *self.component_ids.get_unchecked(bundle_component);
             match storage_type {
                 StorageType::Table => {
-                    let column = table.get_column_mut(component_id).unwrap();
+                    let column =
+                        // SAFETY: If component_id is in self.component_ids, BundleInfo::new requires that
+                        // the target table contains the component.
```

Comment:
It's probably be worth documenting this invariant on the `self.component_ids` field itself.